app/models/authentication.rb

```diff
@@ -6,10 +6,14 @@ class Authentication < ApplicationRecord
   include NewWithTypeStiMixin
   def self.new(*args, &block)
     if self == Authentication
-      AuthUseridPassword.new(*args, &block)
-    else
-      super
+      args = [{}] if args.empty?
+      h = args.first if args.first.kind_of?(Hash)
+
+      type = h[inheritance_column.to_sym] || h[inheritance_column.to_s]
```

Comment:
wouldn't `h` be `nil` if `args.first` is not a Hash?  if so, wouldn't this fail?

Reply:
:facepalm: yes you're right I'll push a change to fix this